siteapp/migrations/0030_auto_20190812_1907.py

```diff
@@ -0,0 +1,54 @@
+# Generated by Django 2.0.13 on 2019-08-12 19:07
+
+from django.db import migrations
+from guardian.shortcuts import assign_perm
+import logging
+logger = logging.getLogger(__name__)
+
+class Migration(migrations.Migration):
+
+    def forwards(self, schema_editor):
+        # method 1
+        Portfolio = apps.get_model('siteapp', 'Portfolio')
+        Project = apps.get_model('siteapp', 'Project')
+
+        projects = Project.objects.all()
+
+        for project in projects:
+            if project.organization and project.portfolio is None:
+                # get or create portfolio from project organization
+                portfolio, created = Portfolio.objects.get_or_create(title=project.organization.name)
+                # assign projects portfolio id
+                project.portfolio = portfolio
+                # save project
+                project.save()
+                # assign portfolio permissions
```

Comment:
I think the approach is correct, but I'm not sure if these role assignments are correct. Is there an easy way to list all the permissions that exist in Guardian?

Reply:
every object has view, change, add and delete - you can also define custom permissions beyond that - owner was a custom permission I defined on portfolio for instance that allows someone to invite another person to have view permissions